server/webapp/WEB-INF/rails.new/app/controllers/api_v1/admin/pluggable_scms_controller.rb

```diff
@@ -28,42 +28,44 @@ def index
 
       def show
         hash = get_scm_hash(params[:material_name])
-        render DEFAULT_FORMAT => hash if stale?(etag: JSON.generate(hash))
+        render DEFAULT_FORMAT => hash if stale?(etag: get_etag_for_scm_object(params[:material_name]))
       end
 
       def create
         result = HttpLocalizedOperationResult.new
         @scm = ApiV1::Scms::PluggableScmRepresenter.new(SCM.new).from_hash(params[:pluggable_scm])
         @scm.ensureIdExists
         pluggable_scm_service.createPluggableScmMaterial(current_user, @scm, result)
-
-        json = ApiV1::Scms::PluggableScmRepresenter.new(@scm).to_hash(url_builder: self)
-        result.isSuccessful ? (render DEFAULT_FORMAT => json) : (render_http_operation_result(result, {data: json}))
+        handle_create_or_update_response(result, @scm)
       end
 
       def update
         result = HttpLocalizedOperationResult.new
-        @scm = ApiV1::Scms::PluggableScmRepresenter.new(SCM.new).from_hash(params[:pluggable_scm])
-        pluggable_scm_service.updatePluggableScmMaterial(current_user, @scm, result)
-
-        json = ApiV1::Scms::PluggableScmRepresenter.new(@scm).to_hash(url_builder: self)
-        result.isSuccessful ? (render DEFAULT_FORMAT => json) : (render_http_operation_result(result, {data: json}))
+        updated_scm = ApiV1::Scms::PluggableScmRepresenter.new(SCM.new).from_hash(params[:pluggable_scm])
+        pluggable_scm_service.updatePluggableScmMaterial(current_user, updated_scm, result, get_etag_for_scm_object(params[:material_name]))
+        handle_create_or_update_response(result, updated_scm)
       end
 
       private
-      def check_for_stale_request
-        if (request.env["HTTP_IF_MATCH"] != "\"#{get_etag_for_material(params[:material_name])}\"")
-          render_message("Someone has modified the global SCM '#{params[:material_name]}'. Please update your copy of the config with the changes.", :precondition_failed)
+      def handle_create_or_update_response(result, updated_scm)
+        if result.isSuccessful
+          json = ApiV1::Scms::PluggableScmRepresenter.new(updated_scm).to_hash(url_builder: self)
+          response.etag = [get_etag_for_scm_object(updated_scm.getName)]
+          render DEFAULT_FORMAT => json
+        else
+          json = ApiV1::Scms::PluggableScmRepresenter.new(updated_scm).to_hash(url_builder: self)
+          render_http_operation_result(result, {data: json})
         end
       end
 
-      def get_etag_for_material(material_name)
-        hash = get_scm_hash(material_name)
-        get_etag_for_scm_json(hash)
+      def check_for_stale_request
```

Comment:
Since we're doing `result.stale` using a generic enough message defined via `STALE_RESOURCE_CONFIG`, lets use `render_result` at both places.

My recommendation earlier was so as to avoid having to declare another internationalization string for something that was used in one place.